core/storage/classifier/gae_models.py

```diff
@@ -181,6 +181,24 @@ def create(
         training_job_instance.put()
         return instance_id
 
+    @classmethod
+    def query_training_jobs(cls, cursor):
```

Comment:
Wouldn't that add one more if condition to check if the length of the `valid_jobs` is greater than 0 and then make the request, if we don't check the length of `valid_jobs` then we will call query_training_jobs even in cases when it is redundant. I think adding comment to cursor should be better.